local-volume/helm/provisioner/templates/provisioner.yaml

```diff
@@ -71,23 +90,15 @@ spec:
             - mountPath: /etc/provisioner/config 
               name: provisioner-config
               readOnly: true
-#
-# Populating entries with user provided disk discovery directories 
-#
-{{- range $tmp, $val := .Values.configmap.storageClass }}
-   {{- range $storageClass, $classConfig := $val }}
-            - mountPath: {{ if $classConfig.mountDir }} {{ $classConfig.mountDir }} {{ else }} {{ $classConfig.hostDir }} {{ end }}
-              name: {{ $storageClass }}
-   {{- end}}
-{{- end}}
+{{- if eq ( $engine | lower ) "gcepre19" }} {{ include "configVolumeMounts" (.Values.configmap.gcePre19.storageClass) | indent 12 }} {{ end }}
+{{- if eq ( $engine | lower ) "gcepost19" }} {{ include "configVolumeMounts" (.Values.configmap.gcePost19.storageClass) | indent 12 }} {{ end }}
+{{- if eq ( $engine | lower ) "gke" }} {{ include "configVolumeMounts" (.Values.configmap.gke.storageClass) | indent 12 }} {{ end }}
+{{- if eq ( $engine | lower ) "baremetal" }} {{ include "configVolumeMounts" (.Values.configmap.baremetal.storageClass) | indent 12 }} {{ end }}
       volumes:
         - name: provisioner-config
           configMap:
             name: {{ $configMapName }}
-{{- range $tmp, $val := .Values.configmap.storageClass }}
-   {{- range $storageClass, $classConfig := $val }}
-        - name: {{ $storageClass }}
-          hostPath:
-            path: {{ $classConfig.hostDir }}
-   {{- end}}
-{{- end}}
+{{- if eq ( $engine | lower ) "gcepre19" }} {{ include "configVolumes" (.Values.configmap.gcePre19.storageClass) | indent 8 }} {{ end }}
+{{- if eq ( $engine | lower ) "gcepost19" }} {{ include "configVolumes" (.Values.configmap.gcePost19.storageClass) | indent 8 }} {{ end }}
+{{- if eq ( $engine | lower ) "gke" }} {{ include "configVolumes" (.Values.configmap.gke.storageClass) | indent 8 }} {{ end }}
+{{- if eq ( $engine | lower ) "baremetal" }} {{ include "configVolumes" (.Values.configmap.baremetal.storageClass) | indent 8 }} {{ end }}
```

Comment:
Where's the service account?

Reply:
https://github.com/kubernetes-incubator/external-storage/blob/master/local-volume/helm/provisioner/templates/provisioner.yaml#L58

Reply:
Thanks i'm blind